### PR TITLE
Feature/integrations

### PR DESCRIPTION
**Pull request by InquestGeronimo**
Updated the SparseML Hugging Face Integrations README for using the SPARSEML CLI and DeepSparse. Content based on our NLP use-case.

### CODE REVIEW

File: integrations/huggingface-transformers/README.md [hunk 1]
```diff
@@ -16,12 +16,8 @@ limitations under the License.
...
-[huggingface/transformers](https://github.com/huggingface/transformers) repository.
-By integrating the robust training flows in the `transformers` repository with the SparseML code base,
-we enable model sparsification techniques on popular NLP models such as [BERT](https://arxiv.org/abs/1810.04805)
-creating smaller and faster deployable versions.
-The techniques include, but are not limted to:
+This directory combines the SparseML recipe-driven approach with the [huggingface/transformers](https://github.com/huggingface/transformers) repository. By integrating the robust training flows in the `transformers` repository with the SparseML code base, we enable model sparsification techniques on popular NLP models such as [BERT](https://arxiv.org/abs/1810.04805) creating smaller and faster deployable versions. The techniques include, but are not limited to:
+
 - Pruning
 - Quantization
 - Pruning and Quantization
```
Comment (mgoin) [inline integrations/huggingface-transformers/README.md:23]:
```suggestion
- Knowledge Distillation
- Layer Dropping
```

File: integrations/huggingface-transformers/README.md [hunk 2]
```diff
@@ -34,87 +30,156 @@ Coming soon!
...
 
-## Quick Tour
+**Note**: Transformers will not immediately install with this command. Instead, a sparsification-compatible version of Transformers will install on the first invocation of the Transformers code in SparseML.
+
+## SparseML CLI
+
+The SparseML installation provides a CLI for sparsifying your models; appending the `--help` argument will provide a full list of options for training in SparseML:
+
+```bash
+sparseml.transformers.question_answering --help
```
Comment (mgoin) [inline integrations/huggingface-transformers/README.md:50]: this is specific to qa in this case, is there a more general base to list here?

File: integrations/huggingface-transformers/README.md [hunk 3]
```diff
@@ -34,87 +30,156 @@ Coming soon!
...
-|----------------------|-----------------------------------------------------------------------------------------------------------------------|
-| recipes              | Typical recipes for sparsifying NLP models along with any downloaded recipes from the SparseZoo.                      |
-| tutorials            | Tutorial walkthroughs for how to sparsify NLP models using recipes.                                                   |
-| transformers         | Integration repository folder used to train and sparsify NLP models (`setup_integration.sh` must run first).            |
-| README.md            | Readme file.                                                                                                          |
-| setup_integration.sh | Setup file for the integration run from the command line.                                                             |
+Once the command has completed, you will have a sparse checkpoint located in `models/sparse_quantized`.
 
-### Exporting for Inference
+### Quantization-Compatible Deployment Environment
```
Comment (mgoin) [inline integrations/huggingface-transformers/README.md:94]: this is not about a deployment environment, it's about getting a quantized model from a sparse base model?

File: integrations/huggingface-transformers/README.md [hunk 4]
```diff
@@ -34,87 +30,156 @@ Coming soon!
...
 
-The DeepSparse Engine [accepts ONNX formats](https://docs.neuralmagic.com/sparseml/source/onnx_export.html) and is engineered to significantly speed up inference on CPUs for the sparsified models from this integration.
-Examples for loading, benchmarking, and deploying can be found in the [DeepSparse repository here](https://github.com/neuralmagic/deepsparse).
+### Exporting to ONNX
+
+The DeepSparse Engine uses the ONNX format to load neural networks and then deliver breakthrough performance for CPUs by leveraging the sparsity and quantization within a network.
+
+You will first export the trained BERT model to an ONNX format. The SparseML installation additionally provided a `sparseml.transformers.export_onnx` command. You will use this to load the training model folder and create a new model.onnx file within. Be sure the `--model_path` argument points to your trained model. By default, it is set to the result from transfer learning a sparse-quantized BERT model: 
+
+    --model_path "models/sparse_quantized".
```
Comment (mgoin) [inline integrations/huggingface-transformers/README.md:126]: this seem extraneous with the command right after

File: integrations/huggingface-transformers/README.md [hunk 5]
```diff
@@ -34,87 +30,156 @@ Coming soon!
...
+from deepsparse.transformers import pipeline
+
+model_path = "zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/12layer_pruned80_quant-none-vnni"
+
+qa_pipeline = pipeline(
+  task="question-answering", 
+  model_path=model_path
+)
+
+inference = qa_pipeline(question="What's my name?", context="My name is Snorlax")
```
Comment (mgoin) [inline integrations/huggingface-transformers/README.md:165]: put a printout so they can see the result by default

File: integrations/huggingface-transformers/README.md (hunk 4 above)
Comment (InquestGeronimo) [reply integrations/huggingface-transformers/README.md:126]: simplified the text

File: integrations/huggingface-transformers/README.md [hunk 6]
```diff
@@ -34,87 +30,161 @@ Coming soon!
...
-When appropriately combined, it becomes possible to create highly sparse and accurate models.
+### Dense Teacher Creation
 
-This integration adds a `--recipe` argument to the [`run_qa.py`](https://github.com/neuralmagic/transformers/blob/master/examples/pytorch/question-answering/run_qa.py) script among others.
-The argument loads an appropriate recipe while preserving the rest of the training pipeline.
-Popular recipes used with this argument are found in the [`recipes` folder](./recipes).
-Distillation arguments to support student-teacher distillation are additionally added to the scripts as they help improve the recovery while sparsifying.
-Otherwise, all other arguments and functionality remain the same as the original repository.
+To enable distillation, you will first create a dense teacher model that the sparse model will learn from while transferring. **If you already have a Transformers-compatible model, you can use this as the dense teacher in place of training one from scratch.** The following command will use the dense BERT base model from the SparseZoo and fine-tune it on the SQuAD dataset, resulting in a model that achieves 0.885 F1 on the validation set:
```
Comment (mgoin) [inline integrations/huggingface-transformers/README.md:69]:
```suggestion
To enable distillation, you will first create a dense teacher model that the sparse model will learn from while transferring. **If you already have a Transformers-compatible model, you can use this as the dense teacher in place of training one from scratch.** The following command will use the dense BERT base model from the SparseZoo and fine-tune it on the SQuAD dataset, resulting in a model that achieves 88.5% F1 on the validation set: 
```

File: integrations/huggingface-transformers/README.md [hunk 7]
```diff
@@ -34,87 +30,161 @@ Coming soon!
...
-| README.md            | Readme file.                                                                                                          |
-| setup_integration.sh | Setup file for the integration run from the command line.                                                             |
+Once the command has completed, you will have a sparse checkpoint located in `models/sparse_quantized`.
 
-### Exporting for Inference
+### Transfer Learn the Model
 
-After sparsifying a model, the `run_qa.py` script can be run with the `--onnx_export_path` argument to convert the model into an [ONNX](https://onnx.ai/) deployment format.
-The export process is modified such that the quantized and pruned models are corrected and folded properly.
+The following command will use the 80% sparse-quantized BERT model from the SparseZoo and fine-tune it on the SQuAD dataset, resulting in a model that achieves an F1 of 0.885 on the validation set. Keep in mind that the `--distill_teacher` argument is set to pull a dense SQuAD model from the SparseZoo to enable it to run independent of the dense teacher step. If you trained a dense teacher, change this out for the path to your model folder:
```
Comment (mgoin) [inline integrations/huggingface-transformers/README.md:99]:
```suggestion
The following command will use the 80% sparse-quantized BERT model from the SparseZoo and fine-tune it on the SQuAD dataset, resulting in a model that achieves an F1 of 88.5% on the validation set. Keep in mind that the `--distill_teacher` argument is set to pull a dense SQuAD model from the SparseZoo to enable it to run independent of the dense teacher step. If you trained a dense teacher, change this out for the path to your model folder:
```

**Review by mgoin — COMMENTED** [review]
lgtm, just nits on the f1 score

**Review by markurtz — APPROVED** [review]

**Review by mgoin — APPROVED** [review]
